src/components/pages/Save/v2/validate.ts

```diff
@@ -1,54 +1,77 @@
-import { State, Reasons, TransactionType } from './types';
+import { SubscribedToken } from '../../../../types';
+import { State, Reasons } from './types';
 
-const validateSave = ({
-  massetState,
-  amount,
-  amountInCredits,
-  transactionType,
-}: State): [false, Reasons] | [true] => {
-  if (!(massetState && massetState.savingsContracts.v1)) {
+export const getInputToken = ({ exchange, massetState}: State): SubscribedToken | undefined => {
```

Comment:
Let's make this simpler for ourselves by perhaps setting an input token and output token (subscribed) in the save provider state; it can be updated similarly to the mAsset state